### ISSUE TITLE
Add GMP Linker Support

### ISSUE THREAD

**Issue by daryakaviani** [cargo-lambda/cargo-lambda#289]
I am attempting to use the [multi-party-ecdsa library](https://github.com/ZenGo-X/multi-party-ecdsa/) library which uses [GMP](https://gmplib.org/) in my Rust lambda function. GMP is a precision arithmetic library which is commonly used in cryptography research, especially via Rust, so its support would be widely useful for the systems, security, and applied cryptography communities.

Running `cargo lambda build` causes the following error:

```
error: linking with /Users/daryakaviani/Library/Caches/cargo-zigbuild/0.11.1/zigcc-x86_64-unknown-linux-gnu.sh
...
ld.lld: error: unable to find library -lgmp
```

This is specific to `cargo-lambda`'s usage of the `zigbuild` linker because `cargo build` works successfully when `gmp` is installed. Also, `cargo zigbuild` works, so it could be the specific way `cargo-lambda` handles environment variables in relation to the `zigbuild` linker. This error is exclusively introduced when I import `multi-party-ecdsa` in `src/main.rs`. To confirm that this is a machine-agnostic issue, I tried it on x86-64 Ubuntu, x86-64 Amazon Linux, x86-64 MacOS, and ARM M1 MacOS and received the same results. I attempted a workaround on Linux by disabling the `zigbuild` linker:

```
cargo lambda build --disable-zig-linker

cargo lambda deploy \
  --iam-role arn:aws:iam::XXXXXXXXXXXX:role/rust_lambda_test \
  func_name

cargo lambda invoke --remote \
--data-ascii '{"params": "some params"}' \
--output-format json \
func_name
```

This successfully compiled my artifacts and deployed the function, but invocation caused:
```
Error: Runtime.ExitError

  × RequestId: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx Error: Runtime exited with error: exit status 1
```

Is it possible for `cargo-zigbuild` or `cargo-lambda` to be updated for GMP support? If there is an existent workaround, please let me know and feel free to move this to Discussions! 

@calavera @messense

**Comment by daryakaviani** [comment]
@calavera Few questions:

1. Are there plans to update to the latest `zigbuild`?
2. Meanwhile, what are the other canonical ways to deploy Rust in a lambda cheaply?
3. Would it be possible for me to obtain a sound binary another way and pair that with `cargo lambda deploy` and `cargo lambda invoke`?

**Comment by daryakaviani** [comment]
> other cargo lambda subcommands will work

@calavera This approach did not work for me. Using `cargo zigbuild` before deploying (even for a simple function that works with `cargo lambda build`) causes issues. Am I going wrong somewhere?
1. Ran `cargo zigbuild --release`. This generated an executable with name `YOUR_PACKAGE_NAME` in the `release` folder.
2. I copied the executable into the folder `target/lambda/YOUR_PACKAGE_NAME` and renamed the executable `bootstrap`.
3. Deployed successfully with `cargo deploy`.
4. Invokation causes the following error:
```
Response
{
  "errorType": "Runtime.ExitError",
  "errorMessage": "RequestId: 744d2237-3c02-4240-8405-aef8e90bf9c0 Error: Runtime exited with error: exit status 1"
}

Function Logs
/var/task/bootstrap: /lib64/libc.so.6: version `GLIBC_2.33' not found (required by /var/task/bootstrap)
/var/task/bootstrap: /lib64/libc.so.6: version `GLIBC_2.28' not found (required by /var/task/bootstrap)
/var/task/bootstrap: /lib64/libc.so.6: version `GLIBC_2.32' not found (required by /var/task/bootstrap)
/var/task/bootstrap: /lib64/libc.so.6: version `GLIBC_2.34' not found (required by /var/task/bootstrap)
/var/task/bootstrap: /lib64/libc.so.6: version `GLIBC_2.33' not found (required by /var/task/bootstrap)
/var/task/bootstrap: /lib64/libc.so.6: version `GLIBC_2.28' not found (required by /var/task/bootstrap)
/var/task/bootstrap: /lib64/libc.so.6: version `GLIBC_2.32' not found (required by /var/task/bootstrap)
/var/task/bootstrap: /lib64/libc.so.6: version `GLIBC_2.34' not found (required by /var/task/bootstrap)
```

**Comment by daryakaviani** [comment]
@calavera Thanks for the update. Unfortunately, this did not seem to fix the problem. `cargo zigbuild` still works, but after updating to 0.13.0, `cargo lambda build` still results in:

```
error: linking with `/home/ubuntu/.cache/cargo-zigbuild/0.14.1/zigcc-x86_64-unknown-linux-gnu.sh` failed: exit status: 1
  |
...
  = note: warning: unsupported linker arg: -znoexecstack
          warning: unsupported linker arg: -zrelro
          warning: unsupported linker arg: -znow
          ld.lld: error: unable to find library -lgmp
```
Are there linking nuances in relation to the zigbuild?

**Comment by daryakaviani** [comment]
> Are you sure you're not using docker or some other host to run zigbuild

@calavera Yep, I am running `cargo zigbuild` in the same directory as `cargo lambda build`. To recreate the issue, you can start from your [basic lambda example](https://github.com/awslabs/aws-lambda-rust-runtime/tree/main/examples/basic-lambda).

In main.rs, import:
```
use multi_party_ecdsa::protocols::multi_party_ecdsa::gg_2020::state_machine::keygen::Keygen;
use round_based::{StateMachine};
```

To Cargo.toml, add to dependencies:
```
round-based = { version = "0.1.4", features = ["dev"] }
multi-party-ecdsa = { version = "0.8.1", git = "https://github.com/ZenGo-X/multi-party-ecdsa.git" }
```

You should see that in both MacOS and Linux, `cargo zigbuild` works but `cargo lambda build` does not work.

**Comment by messense** [comment]
Does settings `RUSTFLAGS='-L my/lib/directory'` env var work?

**Comment by messense** [comment]
IMO zig won't help with the glibc version issue when linking external dylibs since the GMP shared library might require a higher glibc version.
